ios/Capacitor/Capacitor/WebViewAssetHandler.swift

```diff
@@ -25,6 +26,7 @@ open class WebViewAssetHandler: NSObject, WKURLSchemeHandler {
     }
 
     open func webView(_ webView: WKWebView, start urlSchemeTask: WKURLSchemeTask) {
+        pendingTasks.append(urlSchemeTask.hash)
```

Comment:
I'm not 100% sure about what thread this method is called on, but it could potentially be called across threads and you may need to lock access to `pendingTasks` to prevent a crash or data race.

Reply:
yeah, when I changed the Array to Set the app started crashing randomly because of threading, with the Array it never crashed, not sure why.
I've added a thread safe set now.